ember-core/shared/src/main/scala/org/http4s/ember/core/h2/Hpack.scala

```diff
@@ -74,15 +72,15 @@ private[h2] object Hpack extends HpackPlatform {
     tDecoder.decode(is, listener)
     tDecoder.endHeaderBlock()
 
-    val decoded = buffer.toList
-    NonEmptyList.fromList(decoded).toRight(new NoSuchElementException("Header List Was Empty"))
-  }.rethrow
+    val decoded = buffer.result()
+    NonEmptyList.fromListUnsafe(decoded)
+  }
```

Comment:
Oh, forgot to mention: we can use `fromListUnsafe` because we are in a `delay(...)` block anyway, and skip the `.rethrow`.